File: 01_JAVA/38_EX/src/com/lion/school/SoccerStudentClass.java
package com.lion.school;

import java.util.Scanner;

public class SoccerStudentClass extends StudentClass{

    // 골 넣은 횟수
    private int goalCount;

    // 생성자
    public SoccerStudentClass(int studentType){
        // 부모의 생성자를 명시적으로 호출한다.
        super(studentType);
    }

    public void inputStudentInfoBase(Scanner scanner) {
        System.out.println();
        System.out.println("축구부 학생 정보 입력");
        // 부모의 메서드를 호출한다.
        super.inputStudentInfoBase(scanner);
        // 나머지 부분을 입력 받는다.
        System.out.print("넣은 골 개수 : ");
        goalCount = scanner.nextInt();
    }

    @Override
    public void showStudentInfoBase() {
        System.out.println();
        // 부모의 매서드를 호출한다.
        super.showStudentInfoBase();
        // 나머지를 출력한다.
        System.out.printf("넣은 골 개수 : %d\n", goalCount);
    }

    @Override
    public void run() {
        System.out.println("축구선수가 달립니다.");
    }

    // 태클하는 메서드
    public void takkle(){
        System.out.println("태클 한다");
    }

    @Override
    public void doAction() {
        takkle();
    }
}

File: 02_Kotlin/20_EX/src/com/lion/school/SoccerStudentClass.kt
package com.lion.school

import java.util.*

class SoccerStudentClass(var goalCount:Int = 0) : StudentClass() {

    // 학생 정보를 입력받는 메서드
    override fun inputStudentInfo(scanner: Scanner) {
        println()
        println("축구부 학생 정보를 입력해주세요")
        super.inputStudentInfo(scanner)
        print("골 넣은 개수 : ")
        goalCount = scanner.nextInt()
    }

    // 학생 정보를 출력하는 메서드
    override fun printStudentInfo() {
        println()
        println("축구부 학생 정보")
        super.printStudentInfo()
        println("골 넣은 개수 : ${goalCount}골")
    }

    // 달리는 메서드
    override fun run() {
        println("축구 선수가 달립니다")
    }

    // 태클하는 기능
    fun takkle(){
        println("태클을 합니다")
    }
}